security/kubelogin-poc/README.md

```diff
@@ -84,13 +60,149 @@ the generated CA certificate in previous step. Using expired or non-trusted
 certificates causes apiserver not trust Dex, and login attempts will fail even
 if the Dex token kubectl gained is valid.
 
-See [dex.yaml](k8s/dex.yaml).
-
 We also need to add a line to `/etc/hosts` to create fake Dex domain:
-`127.0.0.1       dex.example.com` or simply change all occurences of
+`127.0.0.1       dex.example.com` or simply change all occurrences of
 `dex.example.com` with `127.0.0.1`. Here we use the `dex.example.com` in order
 to make a difference between Dex host and other services.
 
+Depending on the connector and grant type you want to configure, choose the
+appropriate configs below.
+
+#### Password connector
+
+Password flow is enabled only, if Dex config is set to allow it by setting one
+of the connectors as `passwordConnector`:
+
+```yaml
+oauth2:
+  passwordConnector: ldap
+```
+
+If this config is not present, password grant type is rejected as
+`not supported`. Most of the connectors do not support username/password.
+`local` and `ldap` do.
+
+#### Grant types
+
+Kubelogin and Dex support following grant types:
+
+- auto (automatic selection)
+- authcode (token passthrough via web browser callback)
+- authcode-keyboard (generates code you need to paste yourself)
+- device-code (like authcode, but for a device of certain code)
```

Comment:
I'm not sure what this means. Should it be "device of certain **type**"? As in, you have a special device that you use instead of the browser callback.

Reply:
No, it means device = machine you're running kubectl in. Basically kubectl has auth session open, and sits there polling until the Dex gets an auth succeeded from provider that is matched to the device-code of the kubectl session.

I reworded a bit.